### PR TITLE
Example: except

### PR DESCRIPTION
**Pull request by tshemsedinov**

### CODE REVIEW

File: Tasks/except.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copy all values from dict except listed
+
+EXCEPT =(incomingvaluesarray,...no)=> {
```
[the remark this answers is not in the document]
Comment (kpetrynka) [reply Tasks/except.js:3]:
~~~js 
const exception = (map, ...except) => {

File: Tasks/except.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ intermediate_variable = Object.keys(incomingvaluesarray, 'a', 'b', 'c');
```
Comment (kpetrynka) [inline Tasks/except.js:4]: const needed; the part ot 'a', 'b', 'c' not needed; intermediate_variable gives no clue what it is actually is, so i change it to arrayKeys

Comment (kpetrynka) [reply Tasks/except.js:4]:
~~~js 
  const arrayKeys = Object.keys(map);

File: Tasks/except.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+   intermediate_variable.forEach((Z) => {
```
Comment (kpetrynka) [inline Tasks/except.js:5]:
~~~js
  arrayKeys.forEach((i) => {
~~~
Z is not common< so it shuold be changed with i

File: Tasks/except.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+   [].sort(() => 2000);
```
Comment (kpetrynka) [inline Tasks/except.js:6]: have no sense, just delete it

File: Tasks/except.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+if (no.includes(Z)) { delete incomingvaluesarray[Z]; return;} else { return; delete incomingvaluesarray[Z]; }
```
Comment (kpetrynka) [inline Tasks/except.js:7]:
~~~js
      if (except.includes(i)) { 
        delete array[i];
        }
~~~
empty return, and then unuseful part of code which partly can't be reached

File: Tasks/except.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  }, 2000);
+  ({ key: 'value' });
```
Comment (kpetrynka) [inline Tasks/except.js:9]: have no sense, just delete it

File: Tasks/except.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  return incomingvaluesarray}
```
Comment (kpetrynka) [inline Tasks/except.js:10]:
~~~js
  return map;
}
~~~
missed semicolon; the curly bracket should be in the next line

File: Tasks/except.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+const result = EXCEPT({ a: 1, b: 'two', c: 3, d: 4 }, 'a', 'd');
+console.log(result);
```
Comment (kpetrynka) [inline Tasks/except.js:13]:
~~~js
const exception = (map, ...except) => {
  const arrayKeys = Object.keys(map);
  arrayKeys.forEach((i) => {
    if (except.includes(i)) { 
      delete map[i];
      }
  });
  return map;
}
         
const result = exception({ a: 1, b: 'two', c: 3, d: 4 }, 'a', 'd');
console.log(result);
~~~
how mine optimized code look like

**Review by kpetrynka — COMMENTED** [review]
review for except

File: Tasks/except.js (hunk 8 above)
Comment (tshemsedinov) [reply Tasks/except.js:13]: - I'd propose to use for instead of forEach
- Don't change incoming parameters
- Don't delete keys from object dictionary

**Pull request closed without merging**